src/v/security/audit/schemas/utils.h

```diff
@@ -158,12 +159,41 @@ authentication make_authentication_event(
   const request_auth_result& r,
   const ss::sstring& svc_name);
 
+authentication make_authentication_event(
+  const ss::sstring& auth_protocol,
+  const ss::socket_address& local_address,
+  std::string_view service_name,
+  ss::net::inet_address client_addr,
+  uint16_t client_port,
+  std::optional<std::string_view> client_id,
+  authentication::used_cleartext is_cleartext,
+  security::audit::user user);
+
+authentication make_authentication_event(
+  const security::tls::mtls_state& mtls_state,
+  const ss::socket_address& local_address,
+  std::string_view service_name,
+  ss::net::inet_address client_addr,
+  uint16_t client_port,
+  std::optional<std::string_view> client_id);
+
```

Comment:
question: There are a lot of args here. Would this benefit from a arg struct to support future changes?

Reply:
I like this too, we can file that under future enhancements if we don't wanna block this

Reply:
Yeah I have a running list of non-blocking enhancements we can get to once full functionality is done